src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.cs

```diff
@@ -70,8 +74,9 @@ public partial class BlockProcessor : IBlockProcessor
         _rewardCalculator = rewardCalculator ?? throw new ArgumentNullException(nameof(rewardCalculator));
         _blockTransactionsExecutor = blockTransactionsExecutor ?? throw new ArgumentNullException(nameof(blockTransactionsExecutor));
         _receiptsRootCalculator = receiptsRootCalculator ?? ReceiptsRootCalculator.Instance;
+        _blockTree = blockTree ?? throw new ArgumentNullException(nameof(blockTree));
         _beaconBlockRootHandler = new BeaconBlockRootHandler();
-
+        _blockhashStore = new BlockhashStore(_blockTree, _specProvider, _stateProvider);
```

Comment:
Maybe just pass store into constructor instead of creating it here?

Reply:
the idea was that IBlockTree was the reason we needed to add another argument to the constructor and in future if we even need IBlockTree for something again, we wont need to modify the constructor here.

Reply:
okay, will change this